earthaccess/auth.py

```diff
@@ -233,13 +242,15 @@ def get_session(self, bearer_token: bool = True) -> requests.Session:
         Returns:
             class Session instance with Auth and bearer token headers
         """
-        session = SessionWithHeaderRedirection()
+        user, pwd = getattr(self, "username", None), getattr(self, "password", None)
+        auth = (user, pwd) if user and pwd else None
+        session = SessionWithHeaderRedirection(self.system.edl_hostname, auth)
+
         if bearer_token and self.token:
             # This will avoid the use of the netrc after we are logged in
             session.trust_env = False
-            session.headers.update(
-                {"Authorization": f"Bearer {self.token['access_token']}"}
-            )
+            session.headers["Authorization"] = f"Bearer {self.token['access_token']}"
```

Comment:
Is there a difference in update and assign the value directly? this looks better

Reply:
No difference. Just simpler since we're only updating a single entry.